hypertrace-trace-enricher/hypertrace-trace-enricher/src/main/resources/configs/common/application.conf

```diff
@@ -89,6 +89,22 @@ enricher {
       port = ${?ENTITY_SERVICE_PORT_CONFIG}
     }
   }
+
+  EntitySpanEnricher {
```

Comment:
So this has no dependencies, right?

Reply:
An interesting question that I was pondering and never came to a satisfactory answer for - its dependencies are dependent on the definitions of entities from config. So currently, no it has no dependencies - but if I define an entity with an id based on an enriched attribute, then it would have dependencies. Any ideas? The only potential options I see are to:
- add every enricher as a dependency (but new ones will be easily missed)
- Add enrichers as a dependency as part of defining a new attribute projection (again easy to miss, also doesn't support external changes such as exposing an api to add new projections)
- Add some way (or maybe it exists?) to make this enricher run at the end. I was kind of hoping adding it last to the list would do that.

Reply:
Enricher dependency is implemented as a DAG. You can achieve both `1` and `3` just by making this enricher dependent upon the last one in the chain, which doesn't change very often but could change though.

Do you think grouping all span enrichers into a separate group and then running this after that group would simplify the stuff?

Reply:
Also, the names list isn't going to impact the order in which enrichers are run so don't count on being the last one :)

Reply:
> Do you think grouping all span enrichers into a separate group and then running this after that group would simplify the stuff?

Any way of ordering the enrichers would work, but if the order isn't dependent on the listed sequence, how do we ensure any dependency is the last?. My initial thought was to keep the order of the enrichers stable - that is, run them in the order provided in the list, only reordering to enforce dependencies.